core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -51,6 +56,21 @@ public DedupAggregator(DedupConfig dedupConfig) {
     this.dedupConfig = dedupConfig;
   }
 
+  @Override
+  public void prepareRun(BatchAggregatorContext context) throws Exception {
+    super.prepareRun(context);
+    Schema schema = context.getInputSchema();
+    if (schema == null || schema.getFields() == null) {
+      return;
+    }
+
+    FieldOperation operation =
```

Comment:
This is wrong. What this plugin does is to filter the duplicate records based on the unique keys and filter function. The input and output schema basically will not change. So your field lineage operations should be a list of one-to-one field map from one input field to the output field.